File: frontend/src/types/report.ts
export interface Report {
    id: string;

    title: string;
    description: string;

    category: string;

    latitude: number;
    longitude: number;

    address: string;
    landmark: string;

    city: string;
    district: string;
    state: string;

    pincode: string;

    status: string;
}

File: backend/app/schemas/report.py
from uuid import UUID

from pydantic import BaseModel, ConfigDict

from app.models.enums import ReportCategory, ReportStatus


class ReportCreate(BaseModel):
    title: str
    description: str
    category: ReportCategory

    latitude: float
    longitude: float

    address: str
    landmark: str
    city: str
    district: str
    state: str
    pincode: str


class ReportResponse(BaseModel):
    id: UUID

    title: str
    description: str

    category: ReportCategory

    latitude: float
    longitude: float

    address: str
    landmark: str
    city: str
    district: str
    state: str
    pincode: str

    status: ReportStatus

    model_config = ConfigDict(
        from_attributes=True
    )